### DISCUSSION TITLE
Training as a regression model

### DISCUSSION THREAD

**Discussion by aditya-malte** [explosion/spaCy#7007] [Help: Coding & Implementations] [answered]
Hi,
We can train ner, textcat, pos, etc. in spacy (spacy-transformers to be specific). I'd like to know how can a model be trained on regression using spacy. 
From a Huggingface Transformer point of view the solution is pretty simple (just keep loss function as MSE). However, I'm trying to understand how the same can be done in spacy.
Thanks

**Comment by nsorros** [comment]
I would also be keen to try spacy in a regression task so any pointers on how to do it would be super helpful 🙏

**Comment by polm** [comment, marked as answer]
Since there is no built-in component for this you'd need to make one. I don't think we have any plans to make an official component like that at the moment. 

You could try using a binary textcat for this in a pinch, though category confidence is not really equivalent to intensity. Textcat might also be useful as a starting point for a custom component.
